src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +60,592 @@ const ActorInit En_Zod_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/func_80BB01B0.s")
+static AnimationHeader* sAnimations[] = {
+    &gTijoPlayingVivaceAnim, &gTijoReadyToPlayAnim,      &gTijoArmsFoldedAnim,
+    &gTijoPlayingLentoAnim,  &gTijoPlayingAndantinoAnim,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zod/EnZod_Draw.s")
+static Vec3f D_80BB0580 = { 1300.0f, 1100.0f, 0.0f };
+
+void EnZod_Init(Actor* thisx, PlayState* play) {
+    s32 i;
+    EnZod* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 60.0f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gTijoSkel, &gTijoPlayingLentoAnim, this->morphTable, this->JointTable,
+                       OBJECT_ZOD_LIMB_MAX);
+    Animation_PlayLoop(&this->skelAnime, &gTijoPlayingLentoAnim);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->cymbalRots[0] = this->cymbalRots[1] = this->cymbalRots[2] = 0;
+    this->actor.gravity = this->actor.terminalVelocity = -4.0f;
+    this->cymbalRotVels[0] = this->cymbalRotVels[1] = this->cymbalRotVels[2] = 300;
+    this->stateFlags = 0;
+    this->nextAnimIndex = -1;
+    this->curAnimIndex = -1;
+    this->actor.textId = 0;
+    this->unk_298 = 0;
+
+    for (i = 0; i < 6; i++) {
+        this->drumScales[i] = 0.0f;
+        this->drumScaleVels[i] = 0.01;
+    }
+
+    EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+    this->actionFunc = EnZod_PlayDrumsSequence;
+
+    switch (ENZOD_GET_F(thisx)) {
+        case 1:
+            if (gSaveContext.save.weekEventReg[0x4E] & 1) {
+                this->actionFunc = func_80BAFDB4;
+                EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_VIVACE, ANIMMODE_ONCE);
+                this->actor.flags |= ACTOR_FLAG_10;
+                ActorCutscene_SetIntentToPlay((s16)this->actor.cutscene);
+                break;
+            }
+
+            this->actionFunc = func_80BAFB84;
+            if (!(gSaveContext.save.weekEventReg[0x37] & 0x80)) {
+                Actor_MarkForDeath(&this->actor);
+                break;
+            }
+            break;
+
+        case 2:
+            this->actionFunc = func_80BAFF14;
+            this->fogNear = -1;
+            this->stateFlags |= TIJO_STATE_2;
+            break;
+
+        default:
+            if (gSaveContext.save.weekEventReg[0x37] & 0x80) {
+                Actor_MarkForDeath(&this->actor);
+            }
+            this->actor.flags |= ACTOR_FLAG_10;
+            break;
+    }
+}
+
+void EnZod_Destroy(Actor* thisx, PlayState* play) {
+    EnZod* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnZod_RoomConversation(EnZod* this, PlayState* play) {
+    u16 textId;
+
+    if (gSaveContext.save.playerForm != PLAYER_FORM_ZORA) {
+        textId = 0x1227; // Huh? Who are you? You can't just come walkin' in here.
```

Comment:
These look like direct quotes, which we want to avoid. Instead please just summarize the text.